driver/csiplugin/controllerserver.go

```diff
@@ -591,20 +592,91 @@ func (cs *ScaleControllerServer) CreateVolume(ctx context.Context, req *csi.Crea
 		scaleVol.LocalFS = scaleVol.VolBackendFs
 	}
 
+	var RCid string // Remote Cluster ID
+	if volFsInfo.Type == filesystemTypeRemote {
+		glog.V(5).Infof("filesystem %s is remotely mounted, getting cluster ID information of the owning cluster.", volFsInfo.Name)
+		clusterName := strings.Split(volFsInfo.Mount.RemoteDeviceName, ":")[0]
```

Comment:
Shouldn't we check here whether the remote cluster ID is present in clusterMap for this clusterName (and also not expired), before iterating over the list of clusters?